strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManagerFactory.java

```diff
@@ -1,31 +1,45 @@
 package com.orientechnologies.orient.object.jpa;
 
-import javax.persistence.*;
+import javax.persistence.Cache;
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnitUtil;
+import javax.persistence.Query;
+import javax.persistence.SynchronizationType;
 import javax.persistence.criteria.CriteriaBuilder;
 import javax.persistence.metamodel.Metamodel;
+import java.sql.Connection;
+import java.sql.SQLException;
 import java.util.Map;
 import java.util.logging.Logger;
 
-import com.orientechnologies.orient.core.db.OPartitionedDatabasePool;
+import com.google.common.base.Throwables;
+import com.orientechnologies.orient.core.db.ODatabaseDocumentInternal;
+import com.orientechnologies.orient.jdbc.OrientDataSource;
+import com.orientechnologies.orient.jdbc.OrientJdbcConnection;
 import com.orientechnologies.orient.object.db.OObjectDatabaseTx;
 
 /**
  * @author Sergey Bespalov
+ * @author Przemyslaw Fusik
  *
  */
-public class OJPAPartitionedEntityManagerPool implements EntityManagerFactory
+public class OJPAObjectDatabaseTxEntityManagerFactory
+        implements EntityManagerFactory
 {
     /** the log used by this class. */
-    private static Logger logger = Logger.getLogger(OJPAPartitionedEntityManagerPool.class.getName());
+    private static Logger logger = Logger.getLogger(OJPAObjectDatabaseTxEntityManagerFactory.class.getName());
 
-    private OPartitionedDatabasePool databasePool;
+    private final OrientDataSource dataSource;
     private final OJPAProperties properties;
 
-    public OJPAPartitionedEntityManagerPool(final OJPAProperties properties)
+
+    public OJPAObjectDatabaseTxEntityManagerFactory(final OJPAProperties properties,
+                                                    final OrientDataSource dataSource)
     {
         this.properties = properties;
-        this.databasePool = new OPartitionedDatabasePool(properties.getURL(), properties.getUser(),
-                properties.getPassword(), 100, 100);
+        this.dataSource = dataSource;
```

Comment:
@fuss86 what is the difference between `OPartitionedDatabasePool` and `OrientDataSource`? I mean that this guys  seems created too many different `Pool` implementations and I am confused :)

Reply:
>  I mean that this guys seems created too many different Pool implementations and I am confused :)

I'm really confused too :( 

The bad thing is in v. 3.0.0 [OPartitionedDatabasePool#acquire](https://github.com/orientechnologies/orientdb/blob/3.0.0/core/src/main/java/com/orientechnologies/orient/core/db/OPartitionedDatabasePool.java#L176) returns [deprecated ODatabaseDocumentTx](https://github.com/orientechnologies/orientdb/blob/3.0.0/core/src/main/java/com/orientechnologies/orient/core/db/document/ODatabaseDocumentTx.java) instance with comment 

> @Deprecated use {@link OrientDB} instead.

So I started using [OrientDB](https://github.com/orientechnologies/orientdb/blob/3.0.0/core/src/main/java/com/orientechnologies/orient/core/db/OrientDB.java) wherever needed (at the end it showed up that I had to follow this implementation over all the code, right next to [ODatabasePool](https://github.com/orientechnologies/orientdb/blob/3.0.0/core/src/main/java/com/orientechnologies/orient/core/db/ODatabasePool.java) and [OrientDataSource](https://github.com/orientechnologies/orientdb/blob/3.0.0/jdbc/src/main/java/com/orientechnologies/orient/jdbc/OrientDataSource.java)).

OrientDb docs for v.3.0.x [suggests to use ODatabasePool](https://orientdb.com/docs/3.0.x/java/Document-API-Database.html#using-database-pools). I have been digging in their code a while and it looks like they are changing their API a bit but old implementation is still around.

Reply:
> I mean than we need to be sure that it will always provide only one connection within current transaction.

We're providing [only one connection](https://github.com/fuss86/strongbox/blob/e79a791497e5911c67ae450b80afed56d5163db4/strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManagerFactory.java#L78) on entity manager creation, why do you suspect more than one connection ?

Reply:
yes, I mean that we need to be sure that we have only one connection per transaction.

thanks for explanations :)